### PR TITLE
Add option to remove channels from ChannelList

### PR DESCRIPTION
**Pull request by Dominik-Vogel**
Tested and seems to not cause problems - travis test fails... It seems to be a mypy issue. Cannot know if it is a tuple or not.

### CODE REVIEW

**Comment by Dominik-Vogel** [comment]
Hang on I just noticed the test that is not passing here...

**Comment by Dominik-Vogel** [comment]
Thank you @jenshnielsen, this is a great suggestion!

**Comment by jenshnielsen** [comment]
By casting to an array l mean something like
```
channels = cast(List[InstrumentChannel], self._channels)
``` 
Which does not mention array at all, sorry for the confusion

File: qcodes/instrument/channel.py [hunk 1]
```diff
@@ -263,6 +263,18 @@ def append(self, obj: InstrumentChannel):
         self._channel_mapping[obj.short_name] = obj
         return self._channels.append(obj)
 
+    def remove(self, obj: InstrumentChannel):
```
Comment (astafan8) [inline qcodes/instrument/channel.py]: no testing?

Comment (jenshnielsen) [reply qcodes/instrument/channel.py]: It should probably have a test but was not ready to merge anyway. Somehow got pushed to master anyway by accident with failing test

Comment (Dominik-Vogel) [reply qcodes/instrument/channel.py]: I added extensive testing and this should be good to go now, @jenshnielsen, @astafan8

Comment (astafan8) [reply qcodes/instrument/channel.py]: 👍

**Pull request merged**